src/main/java/org/testng/internal/ClassHelper.java

```diff
@@ -172,26 +174,32 @@ private static void logClassNotFoundError(String className, Exception ex) {
   }
 
   public static IJUnitTestRunner createTestRunner(TestRunner runner) {
+    IJUnitTestRunner tr = null;
       try {
           //try to get runner for JUnit 4 first
-          Class.forName("org.junit.Test");
-          IJUnitTestRunner tr = (IJUnitTestRunner) ClassHelper.forName(JUNIT_4_TESTRUNNER).newInstance();
+        Class.forName("org.junit.Test");
+        Class<?> clazz = ClassHelper.forName(JUNIT_4_TESTRUNNER);
+        if (clazz != null) {
+          tr = (IJUnitTestRunner) clazz.newInstance();
           tr.setTestResultNotifier(runner);
-          return tr;
+        }
       } catch (Throwable t) {
-          Utils.log("ClassHelper", 2, "JUnit 4 was not found on the classpath");
+          Utils.log(CLASS_HELPER, 2, "JUnit 4 was not found on the classpath");
           try {
               //fallback to JUnit 3
               Class.forName("junit.framework.Test");
-              IJUnitTestRunner tr = (IJUnitTestRunner) ClassHelper.forName(JUNIT_TESTRUNNER).newInstance();
-              tr.setTestResultNotifier(runner);
-
-              return tr;
+              Class<?> clazz =ClassHelper.forName(JUNIT_TESTRUNNER);
+              if (clazz != null) {
```

Comment:
If clazz is null, I think we should have the testng exception (no sure it was the previous behavior)